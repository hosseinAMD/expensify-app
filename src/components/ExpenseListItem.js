import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = (props) => (
    <div>
        <Link to={`/edit/${props.expense.id}`}><h3>{props.expense.description}</h3></Link>
        <p>{`${props.expense.amount} Rials`}
        --
        {moment(props.expense.createdAt).format('MMMM Do, YYYY')}</p>
    </div>
);

export default ExpenseListItem;