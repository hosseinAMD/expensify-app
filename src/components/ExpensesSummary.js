import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expense-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        <h3>{`Viewing ${props.expensesCount} expenses totalling ${numeral(props.expensesTotal).format('0,0')} Rials`}</h3>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses,state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);