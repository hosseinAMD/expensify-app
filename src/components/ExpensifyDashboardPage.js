import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const ExpensifyDashboardPage = () => (
    <div>
        <p>Dashboard Page</p>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default ExpensifyDashboardPage;