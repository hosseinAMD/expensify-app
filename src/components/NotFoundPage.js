import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <p>404 Page!!! - <Link to="/edit/53">Go Home!</Link></p>
);

export default NotFoundPage;