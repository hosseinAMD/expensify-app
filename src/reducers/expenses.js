//Expenses Reducer

const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
    if(action.type === 'ADD_EXPENSE'){
        return [...state,action.expense];
    } else if(action.type === 'REMOVE_EXPENSE'){
        return state.filter(({id}) => id !== action.id);
    } else if(action.type === 'EDIT_EXPENSE'){
        return state.map((expense) => {
            if(expense.id === action.id){
                return {
                    ...expense,
                    ...action.updates
                };
            } else {
                return expense;
            }
        });
    }
     else {
        return state;
    }
};

export default expensesReducer;