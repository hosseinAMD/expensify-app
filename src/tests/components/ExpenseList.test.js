import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render expenses list',() =>{
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses list with no expenses',() =>{
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});
