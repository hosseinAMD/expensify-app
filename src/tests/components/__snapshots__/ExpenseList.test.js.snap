// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses list 1`] = `
<div>
  <ExpenseListItem
    expense={
      Object {
        "amount": 25000,
        "createdAt": 0,
        "description": "Book",
        "id": "1",
        "note": "",
      }
    }
    key="1"
  />
  <ExpenseListItem
    expense={
      Object {
        "amount": 4000,
        "createdAt": -345600000,
        "description": "Gums",
        "id": "2",
        "note": "",
      }
    }
    key="2"
  />
  <ExpenseListItem
    expense={
      Object {
        "amount": 80000,
        "createdAt": 345600000,
        "description": "Clothes",
        "id": "3",
        "note": "",
      }
    }
    key="3"
  />
</div>
`;

exports[`should render expenses list with no expenses 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
